nilearn/_utils/niimg_conversions.py

```diff
@@ -55,41 +61,62 @@ def check_niimg(niimg, ensure_3d=False):
 
     Its application is idempotent.
     """
+    # If it's an iterator, it's a 4d image
```

Comment:
Shouldn't we say "iterable" instead? A list also exposes the __iter__ attribute for example.